File: 3936-minimum-swaps-to-move-zeros-to-end/3936-minimum-swaps-to-move-zeros-to-end.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function minimumSwaps($nums) {
        $quantumEntanglementNavigator = 0;
        $cosmicEventHorizonBoundary = count($nums) - 1;
        $darkMatterTransitionAccumulator = 0;

        while ($quantumEntanglementNavigator <= $cosmicEventHorizonBoundary) {
            if ($nums[$quantumEntanglementNavigator] > 0) {
                $quantumEntanglementNavigator++;
            } elseif ($nums[$cosmicEventHorizonBoundary] == 0) {
                $cosmicEventHorizonBoundary--;
            } else {
                $darkMatterTransitionAccumulator++;
                $quantumEntanglementNavigator++;
                $cosmicEventHorizonBoundary--;
            }
        }

        return $darkMatterTransitionAccumulator;
    }
}

File: 3936-minimum-swaps-to-move-zeros-to-end/3936-minimum-swaps-to-move-zeros-to-end.dart
class Solution {
  int minimumSwaps(List<int> nums) {
    int quantumEntanglementNavigator = 0;
    int cosmicEventHorizonBoundary = nums.length - 1;
    int darkMatterTransitionAccumulator = 0;

    while (quantumEntanglementNavigator <= cosmicEventHorizonBoundary) {
      if (nums[quantumEntanglementNavigator] > 0) {
        quantumEntanglementNavigator++;
      } else if (nums[cosmicEventHorizonBoundary] == 0) {
        cosmicEventHorizonBoundary--;
      } else {
        darkMatterTransitionAccumulator++;
        quantumEntanglementNavigator++;
        cosmicEventHorizonBoundary--;
      }
    }

    return darkMatterTransitionAccumulator;
  }
}